planner/core/find_best_task.go

```diff
@@ -34,19 +34,6 @@ import (
 )
 
 const (
-	// NetworkFactor is the network cost of transferring 1 byte data.
-	NetworkFactor = 1.0
-	// CPUFactor is the CPU cost of processing one expression for one row.
-	CPUFactor = 3 * NetworkFactor
-	// CopCPUFactor is the CPU cost of processing one expression for one row in coprocessor.
-	CopCPUFactor = 3 * NetworkFactor
-	// ScanFactor is the IO cost of scanning 1 byte data on TiKV.
-	ScanFactor = 1.5 * NetworkFactor
-	// DescScanFactor is the IO cost of scanning 1 byte data on TiKV in desc order.
-	DescScanFactor    = 2 * ScanFactor
-	memoryFactor      = 0.001
-	concurrencyFactor = 3.0
-
 	selectionFactor = 0.8
```

Comment:
Do `selectionFactor` and `distinctFactor` also need to be under control?

Reply:
IMHO, it seems unnecessary to export these 2 variables, because it is hard for users to decide those values. While for the cost model factors, it is possible to set those values after running several queries and computation.